core/src/main/java/org/apache/accumulo/core/security/crypto/AESCBCEncryptionStrategy.java

```diff
@@ -65,16 +71,31 @@ public boolean init(Scope encryptionScope, AccumuloConfiguration conf) throws Ex
   }
 
   @Override
-  public OutputStream encryptStream(OutputStream outputStream) throws Exception {
+  public OutputStream encryptStream(OutputStream outputStream) throws EncryptionStrategyException {
     if (!initialized)
-      throw new Exception("AESEncryptionStrategy not initialized.");
+      throw new EncryptionStrategyException("AESEncryptionStrategy not initialized.");
 
     CryptoUtils.getSha1SecureRandom().nextBytes(initVector);
-    Cipher cipher = Cipher.getInstance(this.transformation);
-    cipher.init(Cipher.ENCRYPT_MODE, skeySpec, new IvParameterSpec(initVector));
+    Cipher cipher;
+    try {
+      cipher = Cipher.getInstance(this.transformation);
+      cipher.init(Cipher.ENCRYPT_MODE, skeySpec, new IvParameterSpec(initVector));
+    } catch (NoSuchAlgorithmException | NoSuchPaddingException | InvalidKeyException
+        | InvalidAlgorithmParameterException e) {
+      throw new EncryptionStrategyException(e);
+    }
 
     CipherOutputStream cos = new CipherOutputStream(outputStream, cipher);
-    cos.write(initVector);
+    try {
+      cos.write(initVector);
+    } catch (IOException e) {
+      try {
```

Comment:
We could put the CipherOutpuStream in a try-with-resources.  This would simplify this try catch.  I can do that after merging if you haven't used it before.

Reply:
That's a better way to do it, and you can have at it.